src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ByLength = 'length',
+  ByAlphabet = 'alphabet',
+  Default = '',
+}
+
+function sortBy(sortType: SortType, isReversed: boolean): string[] {
+  const goods = [...goodsFromServer];
+
+  if (sortType) {
+    switch (sortType) {
+      case SortType.ByAlphabet:
+        goods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SortType.ByLength:
+        goods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.Default);
+  const shouldShowResetButton = sortType || isReversed;
+
+  const preparedGoods: string[] = sortBy(sortType, isReversed);
+
+  const resetAllSort = (): void => {
+    setIsReversed(false);
+    setSortType(SortType.Default);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => {
+            setSortType(SortType.ByAlphabet);
+            sortBy(SortType.ByAlphabet, isReversed);
+          }}
           type="button"
-          className="button is-info is-light"
+          className={
+            classnames('button',
+              'is-info',
+              { 'is-light': sortType !== SortType.ByAlphabet })
+          }
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => {
+            setSortType(SortType.ByLength);
+            sortBy(SortType.ByLength, isReversed);
+          }}
```

Comment:
Let's create a separate handler for this

Reply:
okay)))